apps/spruce/src/pages/image/PackagesTable/index.tsx

```diff
@@ -0,0 +1,140 @@
+  const selectedImage = imageId ?? firstImage;
+  const dispatchToast = useToastContext();
+  const [pagination, setPagination] = useState({
+    pageIndex: 0,
+    pageSize: 10,
+  });
+  const [columnFilters, setColumnFilters] = useState<ColumnFiltersState>([]);
+  const { data: packagesData } = useQuery<
+    ImagePackagesQuery,
+    ImagePackagesQueryVariables
+  >(IMAGE_PACKAGES, {
+    variables: {
+      imageId: selectedImage,
+      opts: {
+        page: pagination.pageIndex,
+        limit: pagination.pageSize,
+        name: columnFilters.find((filter) => filter.id === "name")
+          ?.value as string,
+      },
+    },
+    onError(err) {
+      dispatchToast.error(
+        `There was an error loading image packages: ${err.message}`,
+      );
+    },
+  });
+
+  const packageEntries = useMemo(
+    () => packagesData?.image?.packages.data ?? [],
+    [packagesData?.image?.packages.data],
+  );
+
+  const totalNumPackages = useMemo(
+    () => packagesData?.image?.packages.totalCount ?? 0,
+    [packagesData?.image?.packages.totalCount],
+  );
+
+  const columns: LGColumnDef<Package>[] = [
+    {
+      header: "Name",
+      accessorKey: "name",
+      enableColumnFilter: true,
+      filterFn: "includesString",
+      meta: {
+        search: {
+          "data-cy": "package-name-filter",
+          placeholder: "Search name",
+        },
+        width: "15%",
+      },
+    },
+    {
+      header: "Manager",
+      accessorKey: "manager",
+    },
+    {
+      header: "Version",
+      accessorKey: "version",
+    },
+  ];
+
+  const tableContainerRef = useRef<HTMLDivElement>(null);
+  const table = useLeafyGreenTable<Package>({
+    columns,
+    data: packageEntries,
+    containerRef: tableContainerRef,
+    defaultColumn: {
+      enableColumnFilter: false,
+    },
+    manualPagination: true,
+    manualFiltering: true,
+    rowCount: totalNumPackages,
+    state: {
+      pagination,
+      columnFilters,
+    },
+    onColumnFiltersChange: setColumnFilters,
+    onPaginationChange: setPagination,
+  });
+
+  return (
+    <>
+      <SettingsCardTitle>Packages</SettingsCardTitle>
+      <SettingsCard data-cy="packages-table-card">
+        <BaseTable
+          data-cy-row="packages-table-row"
+          table={table}
+          shouldAlternateRowColor
+        />
+        <PaginationWrapper>
+          <Pagination
+            itemsPerPage={table.getState().pagination.pageSize}
+            onItemsPerPageOptionChange={(value: string) => {
+              table.setPageSize(Number(value));
+            }}
+            numTotalItems={totalNumPackages}
+            currentPage={table.getState().pagination.pageIndex + 1}
+            onCurrentPageOptionChange={(value: string) => {
+              table.setPageIndex(Number(value) - 1);
+            }}
+            onBackArrowClick={() => table.previousPage()}
+            onForwardArrowClick={() => table.nextPage()}
+          />
+        </PaginationWrapper>
```

Comment:
it would be nice if we added this component to the `BaseTable` component so it could be reused for future tables (and then we could remove this code from this component)

Reply:
This seems like a good idea. I'm just a bit worried about how it would impact the BaseTables already in the code. For instance, the User Subscriptions table does pagination on top of the table, while this component does pagination below the table. Would there be a simple way of defining the order of the table and pagination?

Reply:
You can use pass a prop that defines whether to use the pagination comment e.g. `usePagination` and it can be set to `false` by default to not impact any existing tables

Reply:
Got it - added it along with a total count parameter to BaseTable